Classes/Processor/FileProcessor.php

```diff
@@ -95,13 +95,38 @@ public function __construct(
      */
     public function renderFileCollection(string $table, string $fieldName, array $row, array $configuration = []): array
     {
-        $this->configuration = $configuration;
-        $this->setMetaDataConfiguration();
+        $this->initialize($configuration);
 
         $fileCollector = GeneralUtility::makeInstance(FileCollector::class);
         $fileCollector->addFilesFromRelation($table, $fieldName, $row);
         $files = $fileCollector->getFiles();
 
+        return $this->processFileCollection($files);
+    }
+
+    /**
+     * @param array $collection The referencing uid of file collections
```

Comment:
If we accept multiple ids, the variable should be called `$collectionUids` and this comment needs to be adjusted. `The referencing uids of file collections`.